### DISCUSSION TITLE
Decorator that passes an argument?

### DISCUSSION THREAD

**Discussion by probablyjassin** [Pycord-Development/pycord#2634] [Q&A]
I'm trying to make a decorator that passes a variable down to the slash command it's being used on.

```python
# my_decorator.py

def with_myVariable(func):
    @wraps(func)
    async def wrapper(ctx: ApplicationContext, *args, **kwargs):
        custom_channel_var = myVariables.get_channel_variable(ctx.channel.id)
        return await func(ctx, custom_channel_var, *args, **kwargs)

    return wrapper

```

but when i use it and add the variable like this:

```python
# my_slash_command.py

@debug.command(name="test_decorator", description="test")
    @with_myVariable
    async def test_mogi(
        self,
        ctx: ApplicationContext,
        myVariable: str,
    ):
        await ctx.respond(myVariable)
```

pycord thinks it's a variable for user input on the slash command and also gives me this error:
```
discord.errors.ExtensionFailed: Extension 'cogs.debugging' raised an error: TypeError: Invalid class <class 'models.myVariables.myVariable'> used as an input type for an Option
```

**Comment by NeloBlivion** [comment]
Well, all function arguments are interpreted as command options - as such, what you're attempting isn't supported.
Just a guess, but perhaps you would benefit from subclassing `ApplicationContext` and overriding the default `Bot.get_application_context` to implement custom attributes? https://github.com/Pycord-Development/pycord/blob/master/examples/custom_context.py
